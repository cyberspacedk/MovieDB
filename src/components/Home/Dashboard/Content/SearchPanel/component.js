import React from 'react';
import { Form, Field } from 'formik';
import { Layout, Row, Col } from 'antd';
import SearchField from './SearchField/component';

const SearchPanel = () => (
  <Layout.Content>
    <Row type="flex">
      <Col
        xs={{ span: 22, offset: 1 }}
        sm={{ span: 20, offset: 2 }}
        md={{ span: 18, offset: 3 }}
        lg={{ span: 16, offset: 4 }}
        xl={{ span: 14, offset: 5 }}
      >
        <Form>
          <Field name="search" component={SearchField} />
        </Form>
      </Col>
    </Row>
  </Layout.Content>
);

export default SearchPanel;
