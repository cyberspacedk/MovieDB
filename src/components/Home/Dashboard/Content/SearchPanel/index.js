import SearchPanelContainer from './container';

export default SearchPanelContainer;
