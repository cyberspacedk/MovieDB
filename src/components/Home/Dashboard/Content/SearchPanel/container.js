import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import SearchPanel from './component';
import { searchRequest } from '../../../../../store/search/actions';

export const mapPropsToValues = () => ({
  search: '',
});

export const handleSubmit = (values, { props }) => {
  const { search } = values;
  const { searchRequest } = props;
  searchRequest(search);
};

export const validationSchema = Yup.object().shape({
  search: Yup.string()
    .min(3, 'Минимум 3 символа')
    .matches(/^[A-Za-z0-9_-\s]{3,16}$/, 'Только буквы, цифры и тире')
    .required(),
});

const mdtp = {
  searchRequest,
};

export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(SearchPanel);
