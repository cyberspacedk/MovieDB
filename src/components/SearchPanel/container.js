/* eslint-disable no-shadow */
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import SearchPanel from './component';
import { searchRequest } from '../../store/search/actions';

export const mapPropsToValues = () => ({
  search: '',
});

export const handleSubmit = (values, { props, resetForm }) => {
  const { search } = values;
  const { searchRequest } = props;
  resetForm();
  searchRequest(search);
};

export const validationSchema = Yup.object().shape({
  search: Yup.string()
    .min(3, 'Минимум 3 символа')
    .matches(/^[A-Za-z0-9_-]{3,16}$/, 'Только буквы, цифры и тире'),
});
const mdtp = {
  searchRequest,
};
export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(SearchPanel);
