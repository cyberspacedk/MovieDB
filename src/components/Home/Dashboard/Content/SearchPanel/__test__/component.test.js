import React from 'react';
import { shallow } from 'enzyme';
import SearchPanel from '../component';

describe('Component: SearchPanel', () => {
  const container = shallow(<SearchPanel />);

  it('Snapshot: should match. Display search results', () => {
    expect(container).toMatchSnapshot();
  });
});
