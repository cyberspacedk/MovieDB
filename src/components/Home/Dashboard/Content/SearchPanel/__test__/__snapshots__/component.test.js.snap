// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchPanel Snapshot: should match. Display search results 1`] = `
<Adapter>
  <Row
    gutter={0}
    type="flex"
  >
    <Col
      lg={
        Object {
          "offset": 4,
          "span": 16,
        }
      }
      md={
        Object {
          "offset": 3,
          "span": 18,
        }
      }
      sm={
        Object {
          "offset": 2,
          "span": 20,
        }
      }
      xl={
        Object {
          "offset": 5,
          "span": 14,
        }
      }
      xs={
        Object {
          "offset": 1,
          "span": 22,
        }
      }
    >
      <Form>
        <FormikConnect(FieldInner)
          component={[Function]}
          name="search"
        />
      </Form>
    </Col>
  </Row>
</Adapter>
`;
